src/tests/multihost/alltests/test_proxy_rfc2307.py

```diff
@@ -0,0 +1,625 @@
+
+    request.addfinalizer(restore_user_grp)
+
+
+def config_sssd(multihost):
+    """
+    Configure common parameters for sssd.conf
+    """
+    tools = sssdTools(multihost.client[0])
+    sssd_params = {'domains': ds_instance_name}
+    tools.sssd_conf('sssd', sssd_params)
+    domain_name = tools.get_domain_section_name()
+    domain_params = {'debug_level': '9',
+                     'id_provider': 'proxy',
+                     'proxy_lib_name': 'ldap',
+                     'proxy_pam_target': 'sssdproxyldap'}
+    tools.sssd_conf('domain/' + domain_name, domain_params)
+
+
+@pytest.mark.usefixtures('setupds',
+                         'default_sssd',
+                         'sssdproxyldap',
+                         'install_nslcd',
+                         'ldap_objects_sssd_client')
+@pytest.mark.tier1_3
+class TestProxyrfc2307(object):
+    """
+    This is test case class for proxy provider suite
+    """
+    def test_lookup_user_group(self, multihost, backupsssdconf):
+        """
+        :title: case sensitive is true lookup user group
+        :id: b7a0e1b6-4332-11ed-9a72-845cf3eff344
+        :caseposneg: positive
+        :steps:
+          1. While case_sensitive = true, lookup user group
+        :expectedresults:
+          1. Should succeed
+        """
+        tools = sssdTools(multihost.client[0])
+        config_sssd(multihost)
+        tools.clear_sssd_cache()
+        for i in ["id User_CS1 | grep User_CS1_grp1",
+                  "id User_CS1_Alias | grep User_CS1_grp1",
+                  "getent passwd User_CS1 | grep User_CS1",
+                  "getent passwd User_CS1_Alias | grep User_CS1",
+                  "getent group User_CS1_grp1 | grep User_CS1_grp1",
+                  "getent group User_CS1_grp1 | grep User_CS1",
+                  "getent group User_CS1_grp1_Alias | grep User_CS1_grp1",
+                  "getent group User_CS1_grp1_Alias | grep User_CS1"]:
+            execute_cmd(multihost, i)
+        client_hostip = multihost.client[0].ip
+        for user in ['User_CS1', 'User_CS1_Alias']:
+            client = pexpect_ssh(client_hostip, user, 'Secret123', debug=False)
+            try:
+                client.login(login_timeout=30, sync_multiplier=5,
+                             auto_prompt_reset=False)
+            except SSHLoginException:
+                pytest.fail("%s failed to login" % user)
+            else:
+                client.logout()
+        for i in ["getent passwd user_cs1",
+                  "getent passwd user_cs1_alias",
+                  "getent group user_cs1_grp1",
+                  "getent group user_cs1_grp1"]:
+            with pytest.raises(subprocess.CalledProcessError):
+                execute_cmd(multihost, i)
+
+    def test_enumerate_users_groups(self, multihost, backupsssdconf):
+        """
+        :title: enumerate users groups
+        :id: c8ad3f72-4332-11ed-8272-845cf3eff344
+        :caseposneg: positive
```

Comment:
Note, we only need to mention it if the case is negative, default is positive, but it is fine if mentioned.

Reply:
Done